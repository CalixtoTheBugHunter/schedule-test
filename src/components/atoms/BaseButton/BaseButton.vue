<template>
  <component
    :is="tag"
    type="button"
    v-bind="$attrs"
    class="BaseButton"
    :class="BaseButtonClasses"
    :disabled="disabled"
    @click="click"
  >
    <span v-if="!!label" class="BaseButton__Text">
      {{ label }}
    </span>
  </component>
</template>

<script>
export default {
  name: 'BaseButton',
  props: {
    tag: {
      type: String,
      default: 'button',
      validator: val => ['button', 'a'].includes(val)
    },
    label: {
      type: String,
      default: ''
    },
    disabled: {
      type: Boolean,
      default: false
    },
    theme: {
      type: String,
      default: 'primary',
      validator: val => ['primary', 'secondary'].includes(val)
    }
  },
  computed: {
    BaseButtonClasses() {
      return [`BaseButton--${this.theme}`, {
        'BaseButton--disabled': this.disabled
      }]
    }
  },
  methods: {
    click(e) {
      this.$emit('click', e)
    }
  }
}
</script>

<style lang="scss" scoped>
.BaseButton {
  height: 32px;
  -webkit-appearance: none;
  display: flex;
  background: inherit;
  padding: 0 10px;
  cursor: pointer;
  align-items: center;
  position: relative;
  white-space: nowrap;
  border-radius: 5px;
  border: 0 none;
  &:hover {
    filter: brightness(90%) contrast(150%);
  }
  &--primary {
    background: $color-primary;
  }
  &--secondary {
    background: $color-secondary;
  }
  &__Text {
    font-size: 16px;
    color: $color-black;
    text-align: center;
    display: block;
    width: 100%;
  }
  &--disabled {
    pointer-events: none;
    opacity: 0.4;
  }
}
</style>