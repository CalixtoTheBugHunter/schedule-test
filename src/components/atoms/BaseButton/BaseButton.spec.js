/* eslint-disable no-undef */
import { shallowMount } from '@vue/test-utils'
import BaseButton from './BaseButton'

describe('BaseButton', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(BaseButton)
  })

  it('Component exists', () => {
    expect(wrapper.exists()).toBe(true)
  })

  describe('Click in component', () => {

    it('Emit function click', () => {
      wrapper.vm.click()
      expect(wrapper.emitted('click')).toBeTruthy()
    })
  })

  it('Component not changed', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})