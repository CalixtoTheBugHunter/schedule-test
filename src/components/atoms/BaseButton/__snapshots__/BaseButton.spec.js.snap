// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseButton Component not changed 1`] = `
<button type="button" class="BaseButton">
  <!--v-if-->
</button>
`;
